addons/html_editor/static/src/main/media/image_crop_plugin.js

```diff
@@ -58,6 +58,7 @@ export class ImageCropPlugin extends Plugin {
         };
 
         const onSave = () => {
+            this.shared.savePendingImages();
```

Comment:
I'm not sure we want to add such a dependency on `media`. I don't think the plugin responsible from cropping images has to know that there is a pending image mechanism etc. It shouldn't be concerned by that. I think the `saveModifiedImage` method should not crash no matter what. I read the commit message about the src being split but I did not understand what you meant. I wanted to test it on runbot so I could see the traceback in `saveModifiedImage` for myself but I could not reproduce it.

Reply:
@dmo-odoo I have tested and able to reproduce this issue on the 18.0 runbot.

The issue occurs as follows: after cropping an image, it receives a new src reflecting the cropped version. However, when the cropper is reopened, this new src is replaced by the original src. When any cropper button is clicked, a **blur** event triggers the **commitChange** method, which in turn calls **savePendingImages**.

within **savePendingImages**, **saveModifiedImage** is called and attempts to split the src attribute using `el.getAttribute("src").split(":")[1].split(";")[0]`. Since the original src is present due to the cropper, this split operation leads to a traceback error.

here i have saved the modified image after the cropping operation so that next time blur events gets called when cropper is open we dont have to save modified image as its already saved.

Reply:
Thanks for the explanation. Now I understand what is going on. I still think that it is not the concern of the crop plugin to call savePendingImages, however. That being said, I think the problem is that the crop plugin is the one to set the `o_modified_image_to_save` class when it is done cropping, but it does not remove it when it starts cropping again. I think that is the issue. It should remove the class when it starts cropping again (and put it back when cancelling if it was already there.) That way, the saving mechanism will not try to save the image while it is being cropped.

Reply:
Thanks but it should be in `ImageCrop`, not in `ImageCropPlugin`, because that is where the `o_modified_image_to_save` class is added in the first place. Note that when you restore the class, you should only reapply the class if it was there before opening and closing the cropper.

Reply:
@dmo-odoo  Applied the requested changes. can you please take a look.